git-cliff-core/src/repo.rs

```diff
@@ -72,13 +72,21 @@ impl Repository {
 	pub fn commits(
 		&self,
 		range: Option<&str>,
+		include_root: bool,
 		include_path: Option<Vec<Pattern>>,
 		exclude_path: Option<Vec<Pattern>>,
 	) -> Result<Vec<Commit>> {
 		let mut revwalk = self.inner.revwalk()?;
 		revwalk.set_sorting(Sort::TOPOLOGICAL)?;
 		if let Some(range) = range {
-			revwalk.push_range(range)?;
+			// Push only the last range commit if we need to start at root.
+			if include_root {
+				if let Some(dots_index) = range.find("..") {
+					revwalk.push(Oid::from_str(&range[dots_index + 2..])?)?;
+				}
+			} else {
+				revwalk.push_range(range)?;
+			}
```

Comment:
Instead of adding a new flag to the function, we can fallback to using the given `range` string as-is if it does not contain two dot range notation.

e.g.

```rust
if range.contains("..") {
	revwalk.push_range(range)?;
} else {
	revwalk.push(Oid::from_str(&range)?)?;
}
```

Also see my other comment :)

Reply:
Nice, we have an algebraic type (Option) to decide between range and "everything" plus a special value to say "between root and this sha".

I wanted to make CommitRange an enum but then said "surely a boolean would be enough". 😂

Wdyt, is the blast radius of enum too much? If yes, then I'll gladly apply the proposed changes instead.

Reply:
Went with the simple solution for now.

Reply:
Let's keep it simple :)